enhancements/update/distribute-secondary-metadata-as-container-image.md

```diff
@@ -0,0 +1,202 @@
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+
+### Graduation Criteria
+**Note:** *Section not required until targeted at a release.*
+
+- Maturity levels - `Developer Preview`, `Red Hat Staging`, `Red Hat Production`, `GA`
+
+##### `Developer Preview`
+- New Cincinnati plugin
+- Container image registry managed by and dedicated to prow CI
+- Integration tests using an image (to-be-determined: verifying against mocked signatures)
+- CPaaS pipeline is setup up to automatically build and publish a container image for changes to the master branch of the graph-data repository
+
+##### `Developer Preview` -> `Red Hat Staging`
+- Update the Cincinnati deployment template to use the new plugin.
+
+##### Red Hat Staging -> Red Hat Production
+- New plugin is successfully deployed to production
+- Adjust the Cincinnati Operator to use the new plugin
+
+##### Red Hat Production -> GA
+- Document the mirroring process for end-users
+- Definitive positive feedback from the ACM team and their use-case
+
+##### Removing a deprecated feature
+*Not applicable*
+
+### Upgrade / Downgrade Strategy
+*Not applicable*
+
+### Version Skew Strategy
+*Not applicable*
+
+## Implementation History
+
+### Initial implementation
+The initial implementation involves setting up the CPaaS pipeline, implementing a new Cincinnati plugin which downloads the published image and extracts it to disk for subsequent processing in the plugin chain.
+It also involves adjusting the Cincinnati Operator to use the new plugin by default.
+
+## Drawbacks
+
+### Increase OCP release latency
+The enhancement introduces the additional latency of the CPaaS pipeline into the OCP release workflow.
+Assuming that the pipeline is fully automated, the latency introduced is expected to be at the order of minutes.
+This seems to be an acceptable latency with regards to its worst-case scenario, where it's added on top of the duration of blocking an update path via the path of the repository.
+
+## Alternatives
+I want to clearly state that none of the alternatives mentioned below have all the same benefits as the enhancement described here.
+
+* Instead of proceeding with this enhancement, we could continue to work with, or rather around, the current implementation of secondary metadata distribution in Cincinnati.
+
+  Without any code changes, this would mean that we document the various ways how the cincinnati-graph-data GitHub repository can be mirrored and provided to Cincinnati, and how to reconfigure the plugin settings to directly read the data from disk.
+
+  This alternative is currently implemented for Kubernetes deployments in the [Cincinnati Kubernetes operator](https://github.com/openshift/cincinnati-operator).
+  It ships the graph-data using the [init container pattern](https://www.vinsguru.com/cloud-design-pattern-kubernetes-init-container-pattern/).
```

Comment:
s/ships/accesses/

Reply:
Does the init container not contain the data?

Reply:
It does. But the main difference is not in how the container is built or shipped. The difference is primarily in how it is used. It's a matter of using it as an init container vs. retrieving and extracting the data directly from a registry.

Reply:
From the perspective of a user, building and shipping is the relevant action, as the rest is supposedly handled by the cincinnati-operator.

>  It's a matter of using it as an init container vs. retrieving and extracting the data directly from a registry.

"Retrieving" is just another word for "shipping" in this context.
I think that we don't disagree as we both seem to understand the workflows, but I'm not fully getting what you're missing in the proposal. I guessed it would be to explain the implementation details so I added these. Please resolve the thread if this does the trick for you.